src/Elastic.Apm/Metrics/MetricSet.cs

```diff
@@ -27,19 +30,77 @@ public MetricSet(long timestamp, IEnumerable<MetricSample> samples)
 		public SpanInfo Span { get; set; }
 	}
 
-	public class TransactionInfo
+	public class TransactionInfo : IEquatable<TransactionInfo>
```

Comment:
Ideally this could be a `struct`. But this is a `public` type and it'd be a breaking change to change it.

Reply:
Right, but one could argue that implementing `IEquatable` and changing the `Equals` implementation is equally breaking. So we might still consider that and document it.
(We could also make those public types `sealed` in the course of that?)

Reply:
As discussed over zoom, we keep it a `class`, just to be on the safe side. The current change is less risky than making it to a `struct`. At the same time I opened https://github.com/elastic/apm-agent-dotnet/issues/1817 - the change itself is trivial, we can do that once we bump the major version.